src/cc/usdt.cc

```diff
@@ -210,14 +229,44 @@ std::string Context::resolve_bin_path(const std::string &bin_path) {
   return result;
 }
 
-Probe *Context::find_probe(const std::string &probe_name) {
-  for (Probe *p : probes_) {
+std::shared_ptr<Probe> Context::get(const std::string &probe_name) {
```

Comment:
It can. The only reason I'm using `shared_ptr` is to be able to return Probes to use on the test suite. Would you prefer if I used `unique_ptr` and returned bare pointers instead?